src/store/cache/useOutputsCacheStore.ts

```diff
@@ -52,20 +52,24 @@ const createOutputsStore: StateCreator<OutputsStore> = (set, get) => ({
 				outputs = [...currentSession.outputs, ...logs];
 			}
 
-			const isSessionFinished =
-				[SessionStateType.error, SessionStateType.completed, SessionStateType.stopped].includes(
-					sessionStateConverter(sessionInfo.state) as SessionStateType
-				) && !nextPageToken;
+			const isSessionFinished = [
+				SessionStateType.error,
+				SessionStateType.completed,
+				SessionStateType.stopped,
+			].includes(sessionStateConverter(sessionInfo.state) as SessionStateType);
```

Comment:
```suggestion
const finishedStates = new Set([
	SessionStateType.error,
	SessionStateType.completed,
	SessionStateType.stopped,
]);

const isSessionFinished = finishedStates.has(
	sessionStateConverter(sessionInfo.state) as SessionStateType
);
```

This avoids array scanning and reads cleaner.

Reply:
changed